### PR TITLE
GUVNOR-3064, GUVNOR-3065, GUVNOR-3063: Library fixes

### PR DESCRIPTION
**Pull request by paulovmr**
GUVNOR-3064: [Library] Projects created in a custom team with a custom Group ID have wrong Group ID
GUVNOR-3065: [Library] Error after deleting a team and creating a new team with the same name
GUVNOR-3063: [Library] Business Central freezes after opening more assets in a project and deleting the project

### CODE REVIEW

File: kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-backend/src/main/java/org/kie/workbench/common/screens/datamodeller/backend/server/DataModelerServiceImpl.java [hunk 1]
```diff
@@ -830,297 +924,340 @@ public GenerationResult saveModel( final DataModel dataModel,
...
 
     @Override
-    public List<Path> findClassUsages( Path currentPath,
-                                       String className ) {
+    public List<Path> findClassUsages(Path currentPath,
+                                      String className) {
 
         KieProject project = projectService.resolveProject(currentPath);
 
+        if (project == null) {
```
Comment (paulovmr) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-backend/src/main/java/org/kie/workbench/common/screens/datamodeller/backend/server/DataModelerServiceImpl.java]: Real change is the addition this `if` block.

File: kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-backend/src/test/java/org/kie/workbench/common/screens/datamodeller/backend/server/DataModelerServiceTest.java [hunk 2]
```diff
@@ -91,156 +93,168 @@
...
+                              any(org.uberfire.java.nio.file.Path.class),
+                              any(CommentedOption.class));
+        verify(renameHelper,
+               times(1)).postProcess(any(Path.class),
+                                     any(Path.class));
+        verify(ioService,
+               times(1)).endBatch();
+    }
+
+    @Test
```
Comment (paulovmr) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-backend/src/test/java/org/kie/workbench/common/screens/datamodeller/backend/server/DataModelerServiceTest.java]: Real change in this file is this test addition.

**Comment by paulovmr** [comment]
jenkins retest this

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-backend/src/main/java/org/kie/workbench/common/screens/impl/LibraryServiceImpl.java [hunk 3]
```diff
@@ -381,9 +392,15 @@ private Repository getDefaultRepository(final OrganizationalUnit ou) {
             if (repositoryService.getRepository(primaryDefaultRepositoryName) == null) {
                 return createDefaultRepository(ou,
                                                primaryDefaultRepositoryName);
-            } else {
+            } else if (repositoryService.getRepository(secondaryDefaultRepositoryName) == null) {
                 return createDefaultRepository(ou,
                                                secondaryDefaultRepositoryName);
+            } else {
+                int i = 1;
+                while (repositoryService.getRepository(secondaryDefaultRepositoryName + "-" + ++i) != null) {
```
Comment (Rikkola) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-backend/src/main/java/org/kie/workbench/common/screens/impl/LibraryServiceImpl.java]: There is a danger we run out of project names when the size of the int is reached.
( Please, do not take this seriously ;-)

**Pull request merged**